### DISCUSSION TITLE
I am getting an error when using Provider in slate

### DISCUSSION THREAD

**Discussion by jhw2** [react-page/react-page#1241] [Q&A]
The following error occurs when using provider in custom slate
"The `useSlate` hook must be used inside the <Slate> component's context."

How to I use the Provider in slate?

example please....

**Comment by macrozone** [comment]
@jhw2 can you show some code example?

**Reply by jhw2** [reply]
This is my code!

const defaultSlate = slate((def) => ({
...def,
plugins: {
// this will pull in all predefined plugins
...def.plugins,
// you can also add custom plugins. The namespace custom is just for organizing plugins
custom: {
custom1: customSlatePlugin,
katex: katexSlatePlugin,
},
},
}));
export const provideSlate: CellPlugin = {
...defaultSlate,
Renderer: ({ children, data }) => (
<>
{children}
</>
),
Provider: ({ children }) => {
return <>{ children }</>
},
id: 'provideSlate',
title: 'provideSlate',
description: 'Some custom layout plugin',
version: 1,
};

**Reply by macrozone** [reply]
not sure wher the error is coming from, but `provideSlate` can be removed as it does not do anything. defaultSlate is a `CellPlugin`

**Reply by jhw2** [reply]
I try this code but Provider is not working…

const defaultSlate = slate((def) => ({
...def,
Provider: ({ children }) => {
return <>{ children }</>
},
plugins: {
// this will pull in all predefined plugins
...def.plugins,
// you can also add custom plugins. The namespace custom is just for organizing plugins
custom: {
custom1: customSlatePlugin,
katex: katexSlatePlugin,
},
},
}));

**Reply by macrozone** [reply]
remove the Provider, that is not needed like that

**Reply by jhw2** [reply]
Thank for your reply.
I want to add a class to the slate when the cellStyle changes.
